### ISSUE TITLE
Installation failing on Mac OS : [lotus-miner] Error 2

### ISSUE THREAD

**Issue by johanbove** [filecoin-project/homebrew-lotus#11]
Trying fresh install of Lotus through Homebrew, but ends with "Error 2".

System: MacBook Pro (15-inch, 2018)
OS:  11.6.5 (20G527)
CPU: 2,9 GHz 6-Core Intel Core i9 

```bash
 ~ 🖖  $ make --version
GNU Make 3.81

 ~ 🖖  $ brew --version
Homebrew 3.4.3
Homebrew/homebrew-core (git revision 68bfee0df66; last commit 2022-03-22)
Homebrew/homebrew-cask (git revision a6e4519078; last commit 2022-03-22)

 ~ 🖖  $ go version
go version go1.17.8 darwin/amd64
 ~ 🖖  $ brew tap filecoin-project/lotus
Running `brew update --preinstall`...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> Updated Formulae
Updated 2 formulae.

 ~ 🖖  $ brew install lotus
==> Downloading https://github.com/filecoin-project/lotus/archive/refs/tags/v1.14.1.tar.gz
Already downloaded: /Users/jbove/Library/Caches/Homebrew/downloads/444dee257c7ed2c618c92e56c4841e8533919e511ceaa299bf2c6527bc173769--lotus-1.14.1.tar.gz
==> Installing lotus from filecoin-project/lotus
==> git clone --recurse-submodules https://github.com/filecoin-project/filecoin-ffi.git extern/filecoin-ffi
==> git clone --recurse-submodules https://github.com/filecoin-project/serialization-vectors.git extern/serialization-vectors
==> git clone --recurse-submodules https://github.com/filecoin-project/test-vectors.git extern/test-vectors
==> touch build/.update-modules
==> make all
Last 15 lines from /Users/jbove/Library/Logs/Homebrew/lotus/05.make:
[install-filcrypto/main] successfully installed prebuilt libfilcrypto
+ exit
rm -f lotus
rm -f lotus-miner
rm -f lotus-worker
go build  -ldflags="-X=github.com/filecoin-project/lotus/build.CurrentCommit=+git." -o lotus ./cmd/lotus
go build  -ldflags="-X=github.com/filecoin-project/lotus/build.CurrentCommit=+git." -o lotus-worker ./cmd/lotus-seal-worker
go build  -ldflags="-X=github.com/filecoin-project/lotus/build.CurrentCommit=+git." -o lotus-miner ./cmd/lotus-miner
# github.com/lucas-clemente/quic-go/internal/qtls
/Users/jbove/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/lucas-clemente/quic-go@v0.21.2/internal/qtls/go118.go:5:13: cannot use "quic-go doesn't build on Go 1.18 yet." (untyped string constant) as int value in variable declaration
# github.com/lucas-clemente/quic-go/internal/qtls
/Users/jbove/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/lucas-clemente/quic-go@v0.21.2/internal/qtls/go118.go:5:13: cannot use "quic-go doesn't build on Go 1.18 yet." (untyped string constant) as int value in variable declaration
make: *** [lotus] Error 2
make: *** Waiting for unfinished jobs....
make: *** [lotus-miner] Error 2

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
  https://github.com/filecoin-project/homebrew-lotus/issues

 ~ 🖖  $ brew update
Already up-to-date.
```

**Comment by johanbove** [comment]
New try without success after removing the "go" install from Homebrew:

```bash
 brew remove go
Uninstalling /usr/local/Cellar/go/1.18... (11,947 files, 592.6MB)
 ~ 🖖  $ brew cleanup
...
Removing: /Users/jbove/Library/Caches/Homebrew/go_mod_cache... (24,508 files, 633.5MB)
Removing: /Users/jbove/Library/Caches/Homebrew/go_cache... (5,030 files, 427.2MB)
==> This operation has freed approximately 1GB of disk space.
 ~ 🖖  $ brew install lotus
==> Downloading https://ghcr.io/v2/homebrew/core/go/manifests/1.18
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/go/blobs/sha256:5dfda57937f94683cbe1b31e10b7ed4d0661722ece896dca9cb0c17bbbeaa52a
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:5dfda57937f94683cbe1b31e10b7ed4d0661722ece896dca9cb0c17bbbeaa52a?se=2022
######################################################################## 100.0%
==> Downloading https://github.com/filecoin-project/lotus/archive/refs/tags/v1.14.1.tar.gz
Already downloaded: /Users/jbove/Library/Caches/Homebrew/downloads/444dee257c7ed2c618c92e56c4841e8533919e511ceaa299bf2c6527bc173769--lotus-1.14.1.tar.gz
==> Installing lotus from filecoin-project/lotus
==> Installing dependencies for filecoin-project/lotus/lotus: go
==> Installing filecoin-project/lotus/lotus dependency: go
==> Pouring go--1.18.big_sur.bottle.tar.gz
🍺  /usr/local/Cellar/go/1.18: 11,947 files, 592.6MB
==> Installing filecoin-project/lotus/lotus
==> git clone --recurse-submodules https://github.com/filecoin-project/filecoin-ffi.git extern/filecoin-ffi
==> git clone --recurse-submodules https://github.com/filecoin-project/serialization-vectors.git extern/serialization-vectors
==> git clone --recurse-submodules https://github.com/filecoin-project/test-vectors.git extern/test-vectors
==> touch build/.update-modules
==> make all
Last 15 lines from /Users/jbove/Library/Logs/Homebrew/lotus/05.make:
go: downloading github.com/jackpal/go-nat-pmp v1.0.2
go: downloading github.com/mikioh/tcpopt v0.0.0-20190314235656-172688c1accc
go: downloading github.com/marten-seemann/qtls-go1-17 v0.1.0-rc.1
go: downloading github.com/mikioh/tcpopt v0.0.0-20190314235656-172688c1accc
go: downloading github.com/marten-seemann/qtls-go1-17 v0.1.0-rc.1
go: downloading google.golang.org/genproto v0.0.0-20210917145530-b395a37504d4
go: downloading google.golang.org/genproto v0.0.0-20210917145530-b395a37504d4
go: downloading google.golang.org/genproto v0.0.0-20210917145530-b395a37504d4
# github.com/lucas-clemente/quic-go/internal/qtls
/Users/jbove/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/lucas-clemente/quic-go@v0.21.2/internal/qtls/go118.go:5:13: cannot use "quic-go doesn't build on Go 1.18 yet." (untyped string constant) as int value in variable declaration
# github.com/lucas-clemente/quic-go/internal/qtls
/Users/jbove/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/lucas-clemente/quic-go@v0.21.2/internal/qtls/go118.go:5:13: cannot use "quic-go doesn't build on Go 1.18 yet." (untyped string constant) as int value in variable declaration
make: *** [lotus] Error 2
make: *** Waiting for unfinished jobs....
make: *** [lotus-miner] Error 2

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
  https://github.com/filecoin-project/homebrew-lotus/issues

These open issues may also help:
Installation failing on Mac OS : [lotus-miner] Error 2 https://github.com/filecoin-project/homebrew-lotus/issues/11
```

**Comment by dkkapur** [comment]
@coryschwartz any thoughts on this one? getting blocked on some on-chain analysis work for Fil+.

**Comment by Clete2** [comment]
Same issue on MacBook Pro 15" 2017.

**Comment by smagdali** [comment]
adding a +1 to this... confirmed on a 2021 14" Pro.

**Comment by smagdali** [comment]
Looks like this is the same problem over at IPFS https://github.com/ipfs/go-ipfs/issues/8819, which has now been fixed.

Actually, I think I'm seeing a different error- quic doesn't seem to be the culprit. Gonna open a new issue

**Comment by smagdali** [comment]
@tiancid amd @Clete2 do you see this error or https://github.com/filecoin-project/homebrew-lotus/issues/13 ?

**Comment by ianconsolata** [comment]
The new homebrew approach no longer builds from source, it downloads a prebuilt binary so this should no longer be an issue.
